Engine/BaseAssets/Components/SkeletalMeshComponent.cs

```diff
@@ -0,0 +1,295 @@
+                    Format = SharpDX.DXGI.Format.Unknown,
+                    Buffer = new ShaderResourceViewDescription.BufferResource()
+                    {
+                        FirstElement = 0,
+                        ElementCount = BonesTransformationsCount
+                    }
+                });
+                inverseTransposeBonesBuffer = SharpDX.Direct3D11.Buffer.Create(GraphicsCore.CurrentDevice, BindFlags.ShaderResource,
+                                            InverseTransposeBonesTransformations.ToArray(), matrixSize * BonesTransformationsCount,
+                                            ResourceUsage.Dynamic, CpuAccessFlags.Write,
+                                            ResourceOptionFlags.BufferStructured, matrixSize);
+                inverseTransposeBonesResourceView = new ShaderResourceView(GraphicsCore.CurrentDevice, inverseTransposeBonesBuffer, new ShaderResourceViewDescription()
+                {
+                    Dimension = SharpDX.Direct3D.ShaderResourceViewDimension.Buffer,
+                    Format = SharpDX.DXGI.Format.Unknown,
+                    Buffer = new ShaderResourceViewDescription.BufferResource()
+                    {
+                        FirstElement = 0,
+                        ElementCount = BonesTransformationsCount
+                    }
+                });
+            }
+        }
+
+        private void Use(List<Matrix4x4f> BonesTransformations, List<Matrix4x4f> InverseTransposeBonesTransformations)
+        {
+            if (bonesBuffer is not null && bonesResourceView is not null)
+            {
+                DataStream dataStream;
+                GraphicsCore.CurrentDevice.ImmediateContext.MapSubresource(bonesBuffer, MapMode.WriteDiscard, MapFlags.None, out dataStream);
+                for (int i = 0; i < BonesTransformationsCount; i++)
+                    dataStream.Write(BonesTransformations[i]);
+                GraphicsCore.CurrentDevice.ImmediateContext.UnmapSubresource(bonesBuffer, 0);
+                GraphicsCore.CurrentDevice.ImmediateContext.VertexShader.SetShaderResource(0, bonesResourceView);
+            }
+            if (inverseTransposeBonesBuffer is not null && inverseTransposeBonesResourceView is not null)
+            {
+                DataStream dataStream;
+                GraphicsCore.CurrentDevice.ImmediateContext.MapSubresource(inverseTransposeBonesBuffer, MapMode.WriteDiscard, MapFlags.None, out dataStream);
+                for (int i = 0; i < BonesTransformationsCount; i++)
+                    dataStream.Write(InverseTransposeBonesTransformations[i]);
+                GraphicsCore.CurrentDevice.ImmediateContext.UnmapSubresource(inverseTransposeBonesBuffer, 0);
+                GraphicsCore.CurrentDevice.ImmediateContext.VertexShader.SetShaderResource(1, inverseTransposeBonesResourceView);
+            }
+        }
+
+        private void CheckBonesTransformationsCount(List<Matrix4x4f> BonesTransformations)
```

Comment:
this method is called both at the end of UpdateAnimation method and EnsureGpuBuffer method, which is basically 2 calls in a row - what for?